<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement" />


<title>Experimental Design I: replicatie en power</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>


<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Biostatistiek 2021</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Hoorcolleges
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="introduction.html">Introductie</a>
    </li>
    <li>
      <a href="10-MultipleRegression.html">Algemeen Lineair Model</a>
    </li>
    <li>
      <a href="10-MultipleRegression-FactorialDesigns.html">Factoriële Designs</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    Labs
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ttest_example.html">Practicum 1: Voorbeeld two-sample t-test</a>
    </li>
    <li>
      <a href="anova_example.html">Practicum 1: Voorbeeld anova</a>
    </li>
    <li>
      <a href="Practicum_1_Voorbereiding_1.html">Practicum 1: Voorbereiding 1</a>
    </li>
    <li>
      <a href="Practicum_1_Voorbereiding_2.html">Practicum 1: Voorbereiding 2</a>
    </li>
    <li>
      <a href="Practicum_1_Voorbereiding_3.html">Practicum 1: Voorbereiding 3</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/statOmics/biostatistics21">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="http://statomics.github.io/">statOmics</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Experimental Design I: replicatie en power</h1>
<h4 class="author">Lieven Clement</h4>
<h4 class="date">statOmics, Ghent University (<a href="https://statomics.github.io" class="uri">https://statomics.github.io</a>)</h4>

</div>


<p><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">library</span>(tidyverse)</span></code></pre></div>
<pre><code>## ── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>## ✔ ggplot2 3.3.5     ✔ purrr   0.3.4
## ✔ tibble  3.1.4     ✔ dplyr   1.0.7
## ✔ tidyr   1.1.3     ✔ stringr 1.4.0
## ✔ readr   1.4.0     ✔ forcats 0.5.1</code></pre>
<pre><code>## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()</code></pre>
<div id="concepts" class="section level1">
<h1><span class="header-section-number">1</span> Concepts</h1>
<ul>
<li>Experimental units representative for population: Randomisation</li>
<li>Replication: technical vs biological, sample size - power</li>
<li>Sources of variation: technical, biological, within subject, between subject</li>
</ul>
</div>
<div id="replicatie" class="section level1">
<h1><span class="header-section-number">2</span> Replicatie</h1>
<p>Two pager on Replication in nature methods [<a href="https://www.nature.com/articles/nmeth.3091.pdf">PDF</a>] <embed src="https://www.nature.com/articles/nmeth.3091.pdf" style="width:100.0%" /></p>
<div id="example" class="section level2">
<h2><span class="header-section-number">2.1</span> Example</h2>
<p>Consider a single cell RNA-seq experiment where researchers want to assess the effect of drug treatment on gene expression</p>
<p>Potential research questions</p>
<ul>
<li>Is there a difference in average gene expression between treated and non treated samples</li>
<li>Is there a difference in variability of gene expression between treated and non treated samples</li>
</ul>
</div>
<div id="sources-of-variation" class="section level2">
<h2><span class="header-section-number">2.2</span> Sources of variation</h2>
<p>TABLE 1 NATURE METHODS | VOL.11 NO.9 | SEPTEMBER 2014 | 879 - 880</p>
<table>
<colgroup>
<col width="35%" />
<col width="28%" />
<col width="35%" />
</colgroup>
<thead>
<tr class="header">
<th align="center"></th>
<th align="left">Replicate type</th>
<th align="center">Replicate category<span class="math inline">\(^\text{a}\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"></td>
<td align="left">Colonies</td>
<td align="center">B</td>
</tr>
<tr class="even">
<td align="center"></td>
<td align="left">Strains</td>
<td align="center">B</td>
</tr>
<tr class="odd">
<td align="center">Animal study subjects</td>
<td align="left">Cohoused groups</td>
<td align="center">B</td>
</tr>
<tr class="even">
<td align="center"></td>
<td align="left">Gender</td>
<td align="center">B</td>
</tr>
<tr class="odd">
<td align="center"></td>
<td align="left">Individuals</td>
<td align="center">B</td>
</tr>
<tr class="even">
<td align="center"></td>
<td align="left"></td>
<td align="center"></td>
</tr>
<tr class="odd">
<td align="center"></td>
<td align="left">Organs from sacrificed animals</td>
<td align="center">B</td>
</tr>
<tr class="even">
<td align="center"></td>
<td align="left">Methods for dissociating cells from tissue</td>
<td align="center">T</td>
</tr>
<tr class="odd">
<td align="center">Sample preparation</td>
<td align="left">Dissociation runs from given tissue sample</td>
<td align="center">T</td>
</tr>
<tr class="even">
<td align="center"></td>
<td align="left">Individual cells</td>
<td align="center">B</td>
</tr>
<tr class="odd">
<td align="center"></td>
<td align="left"></td>
<td align="center"></td>
</tr>
<tr class="even">
<td align="center"></td>
<td align="left">RNA-seq library construction</td>
<td align="center">T</td>
</tr>
<tr class="odd">
<td align="center"></td>
<td align="left">Runs from the library of a given cell</td>
<td align="center">T</td>
</tr>
<tr class="even">
<td align="center">Sequencing</td>
<td align="left">Reads from different transcript molecules</td>
<td align="center">V<span class="math inline">\(^\text{b}\)</span></td>
</tr>
<tr class="odd">
<td align="center"></td>
<td align="left">Reads with unique molecular identifier (UMI) from a given transcript molecule</td>
<td align="center">T</td>
</tr>
</tbody>
</table>
<ol style="list-style-type: lower-alpha">
<li>Replicates are categorized as biological (B), technical (T) or of variable type (V).</li>
<li>Sequence reads serve diverse purposes depending on the application and how reads are used in analysis.</li>
</ol>
<hr />
</div>
<div id="at-which-level-should-we-replicate" class="section level2">
<h2><span class="header-section-number">2.3</span> At which level should we replicate?</h2>
<ul>
<li><span class="math inline">\(\text{var}(X)=\sigma^2_A+\sigma^2_C+\sigma^2_M=\sigma^2_{TOT}\)</span></li>
<li><span class="math inline">\(\text{var}(\bar{X})=\frac{\sigma^2_A}{n_A}+\frac{\sigma^2_C}{n_A n_C} + \frac{\sigma^2_M}{n_A n_C n_M}\)</span></li>
</ul>
<p><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fnmeth.3091/MediaObjects/41592_2014_Article_BFnmeth3091_Fig1_HTML.jpg" /> Figure 1 NATURE METHODS | VOL.11 NO.9 | SEPTEMBER 2014 | 879 - 880</p>
<ol style="list-style-type: lower-alpha">
<li>Three levels of replication (two biological, one technical) with animal, cell and measurement replicates normally distributed with a mean across animals of 10 and ratio of variances 1:2:0.5. Solid green (biological) and blue (technical) dots show how a measurement of the expression (X = 12) samples from all three sources of variation. Distribution s.d. is shown as horizontal lines.</li>
<li>Expression variance, Var(X), and variance of expression mean, Var(<span class="math inline">\(\bar X\)</span>), computed across 10,000 simulations of nAnCnM = 48 measurements for unique combinations of the number of animals (nA = 1 to 48), cells per animal (nC = 1 to 48) and technical replicate measurements per cell (nM = 1 and 3). The ratio of Var(X) and Var(<span class="math inline">\(\bar X\)</span>) is the effective sample size, n, which corresponds to the equivalent number of statistically independent measurements. Horizontal dashed lines correspond to biological and total variation. Error bars on Var(X) show s.d. from the 10,000 simulated samples (nM = 1).</li>
</ol>
<hr />
<ul>
<li>Cost benefit trade-off: cost difference between biological and technical replicates</li>
<li>Typically, biological variability is substantially larger than technical variability</li>
</ul>
<p><span class="math inline">\(\rightarrow\)</span> commit resources to sampling biologically relevant variables</p>
<p><span class="math inline">\(\rightarrow\)</span> unless measures of technical variability are of interest then increasing number of technical measurements is valuable.</p>
<ul>
<li><p>Good experimental design practice includes planning for replication.</p>
<ol style="list-style-type: decimal">
<li>Identify the research questions to be answered with experiments.</li>
<li>Determine proportion of variability induced by each step.</li>
<li>Be aware for pseudoreplication and aim for statistically independent replicates.</li>
</ol></li>
</ul>
<hr />
</div>
</div>
<div id="power-sample-size-and-other-design-aspects" class="section level1">
<h1><span class="header-section-number">3</span> Power, sample size and other design aspects</h1>
<p>Reading materials: <a href="https://www.nature.com/articles/nmeth.2738.pdf">Nature Methods (2013), 10(12), 1139–1140</a></p>
<div id="intermezzo-linear-regression-in-matrix-form" class="section level2">
<h2><span class="header-section-number">3.1</span> Intermezzo linear regression in matrix form</h2>
<ul>
<li>Linear regression is a very important statistical tool to study the association between variables and to build prediction models.</li>
</ul>
<div id="scalar-form" class="section level3">
<h3><span class="header-section-number">3.1.1</span> Scalar form</h3>
<ul>
<li>Consider a vector of predictors <span class="math inline">\(\mathbf{x}=(x_1,\ldots,x_p)\)</span> and</li>
<li>a real-valued response <span class="math inline">\(Y\)</span></li>
<li>then the linear regression model can be written as <span class="math display">\[
Y=f(\mathbf{x}) +\epsilon=\beta_0+\sum\limits_{j=1}^p x_j\beta_j + \epsilon
\]</span> with i.i.d. <span class="math inline">\(\epsilon\sim N(0,\sigma^2)\)</span></li>
</ul>
</div>
<div id="vectormatrix-form" class="section level3">
<h3><span class="header-section-number">3.1.2</span> Vector/Matrix form</h3>
<ul>
<li><span class="math inline">\(n\)</span> observations <span class="math inline">\((\mathbf{x}_1,y_1) \ldots (\mathbf{x}_n,y_n)\)</span> with <span class="math inline">\(\mathbf{x}_1^T=[1 x_1 \ldots x_p]\)</span></li>
<li>Regression in matrix notation <span class="math display">\[\mathbf{Y}=\mathbf{X\beta} + \mathbf{\epsilon}\]</span> with <span class="math inline">\(\mathbf{Y}=\left[\begin{array}{c}y_1\\ \vdots\\y_n\end{array}\right]\)</span>, <span class="math inline">\(\mathbf{X}=\left[\begin{array}{cccc} 1&amp;x_{11}&amp;\ldots&amp;x_{1p}\\ \vdots&amp;\vdots&amp;&amp;\vdots\\ 1&amp;x_{n1}&amp;\ldots&amp;x_{np} \end{array}\right]\)</span> or <span class="math inline">\(\mathbf{X}=\left[\begin{array}{c} \mathbf{x}_1^T\\\vdots\\\mathbf{x}_n^T\end{array}\right]\)</span>, <span class="math inline">\(\boldsymbol{\beta}=\left[\begin{array}{c}\beta_0\\ \vdots\\ \beta_p\end{array}\right]\)</span> and <span class="math inline">\(\mathbf{\epsilon}=\left[\begin{array}{c} \epsilon_1 \\ \vdots \\ \epsilon_n\end{array}\right]\)</span></li>
</ul>
<hr />
</div>
</div>
<div id="least-squares-ls" class="section level2">
<h2><span class="header-section-number">3.2</span> Least Squares (LS)</h2>
<ul>
<li><p>Minimize the residual sum of squares <span class="math display">\[\begin{eqnarray*}
RSS(\boldsymbol{\beta})&amp;=&amp;\sum\limits_{i=1}^n e^2_i\\
&amp;=&amp;\sum\limits_{i=1}^n \left(y_i-\beta_0-\sum\limits_{j=1}^p x_{ij}\beta_j\right)^2
\end{eqnarray*}\]</span></p></li>
<li><p>or in matrix notation</p></li>
</ul>
<p><span class="math display">\[
\text{RSS}(\boldsymbol{\beta})=(\mathbf{Y}-\mathbf{X\beta})^T(\mathbf{Y}-\mathbf{X\beta})
\]</span></p>
<p><span class="math display">\[\rightarrow \hat{\boldsymbol{\beta}}=\text{argmin}_\beta \text{ RSS}(\boldsymbol{\beta})\]</span></p>
<hr />
<div id="minimize-rss" class="section level3">
<h3><span class="header-section-number">3.2.1</span> Minimize RSS</h3>
<p><span class="math display">\[
\begin{array}{ccc}
\frac{\partial RSS}{\partial \boldsymbol{\beta}}&amp;=&amp;\mathbf{0}\\\\
\frac{(\mathbf{Y}-\mathbf{X\beta})^T(\mathbf{Y}-\mathbf{X}\boldsymbol{\beta})}{\partial \boldsymbol{\beta}}&amp;=&amp;\mathbf{0}\\\\
-2\mathbf{X}^T(\mathbf{Y}-\mathbf{X}\boldsymbol{\beta})&amp;=&amp;\mathbf{0}\\\\
\mathbf{X}^T\mathbf{X\beta}&amp;=&amp;\mathbf{X}^T\mathbf{Y}\\\\
\hat{\boldsymbol{\beta}}&amp;=&amp;(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{Y}
\end{array}
\]</span></p>
<hr />
</div>
</div>
<div id="variance-estimator" class="section level2">
<h2><span class="header-section-number">3.3</span> Variance Estimator?</h2>
<p><span class="math display">\[
\begin{array}{ccl}
\hat{\boldsymbol{\Sigma}}_{\hat{\boldsymbol{\beta}}}
&amp;=&amp;\text{var}\left[(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{Y}\right]\\\\
&amp;=&amp;(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\text{var}\left[\mathbf{Y}\right]\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}\\\\
&amp;=&amp;(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T(\mathbf{I}\sigma^2)\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}
\\\\
&amp;=&amp;(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{I}\quad\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}\sigma^2\\\\
%\hat{\boldmath{\Sigma}}_{\hat{\boldsymbol{\beta}}}&amp;=&amp;(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T \text{var}\left[\mathbf{Y}\right](\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}\\
&amp;=&amp;(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}\sigma^2\\\\
&amp;=&amp;(\mathbf{X}^T\mathbf{X})^{-1}\sigma^2
\end{array}
\]</span></p>
<p>The uncertainty on the model parameters thus depends on the residual variability and the design!</p>
<ul>
<li>The larger <span class="math inline">\(\mathbf{X}^T\mathbf{X}\)</span> the more information the experiment contains on the model parameters and the smaller their variances and standard errors will be!</li>
<li>Factorial designs?</li>
<li>Designs with continuous predictors?</li>
</ul>
<hr />
<p>The effect size of interest is typically a linear combinations of the model parameters, i.e.  <span class="math display">\[
l_0 \times \beta_0 + l_1 \times \beta_1 + ... + l_{p-1} \times \beta_{p-1} = \mathbf{L}^T\boldsymbol{\beta}
\]</span></p>
<p>The null hypothesis of our test statistics is than formulated as</p>
<p><span class="math display">\[
H_0: \mathbf{L}^T\boldsymbol{\beta} = 0 
\]</span></p>
<p>vs the alternative hypothesis</p>
<p><span class="math display">\[
H_0: \mathbf{L}^T\boldsymbol{\beta} \neq 0 
\]</span> which can be assessed using a t-test statistic:</p>
<p><span class="math display">\[
t=\frac{\mathbf{L}^T\hat{\boldsymbol{\beta}} - 0}{\text{se}_{\mathbf{L}^T\hat{\boldsymbol{\beta}}}}
\]</span> which follows a t-distribution with n-p degrees of freedom under the null hypothesis when all assumptions for the linear model are valid.</p>
<p>So the power</p>
<p><span class="math display">\[P(p &lt; 0.05 | H_1)\]</span></p>
<p>will typically depends on</p>
<ul>
<li>the real effect size in the population <span class="math inline">\(\mathbf{L}^T\boldsymbol{\beta}\)</span>.</li>
<li>Number of observations: SE and df of t-test.</li>
<li>Choice of the design points.</li>
<li>Choice of significance level.</li>
</ul>
<p>Similar to the introductory example, we can use simulations to assess the power.</p>
</div>
<div id="mouse-example" class="section level2">
<h2><span class="header-section-number">3.4</span> Mouse example</h2>
<ul>
<li><p>In 2021 Choa et al. published that the cytokine Thymic stromal lymphopoietin (TSLP) induced fat loss through sebum secretion (talg). [<a href="https://www.science.org/doi/full/10.1126/science.abd2893">html</a>] [<a href="https://www.science.org/doi/pdf/10.1126/science.abd2893">PDF</a>]</p></li>
<li><p>Suppose that you would like to set up a similar study to test if cytokine interleukin 25 (IL) also has beneficial effect.</p></li>
<li><p>You plan to setup a study with a control group of high fat diet (HFD) fed mice and a treatment group that recieves the HFD and IL.</p></li>
<li><p>What sample size do you need to pick up the effect of the treatment.</p></li>
</ul>
<div id="how-will-we-analyse-the-data-of-this-experiment" class="section level3">
<h3><span class="header-section-number">3.4.1</span> How will we analyse the data of this experiment?</h3>
<hr />
<ul>
<li><p><span class="math inline">\(H_0\)</span>: The average weight difference is equal to zero</p></li>
<li><p><span class="math inline">\(H-0\)</span>: The average weight difference is different from zero</p></li>
<li><p>Two sample t-test or a t-test on the slope of a linear model with one dummy variable.</p></li>
</ul>
<p><span class="math display">\[ Y_i = \beta_0 + \beta_1 X_\text{iIL} + \epsilon_i\]</span></p>
<p><span class="math display">\[
\text{with }
X_\text{iIL}=\begin{cases}X_{iIL}=0 &amp; \text{HFD}\\X_{iIL}=1 &amp; \text{HFD + IL}\end{cases}.
\]</span></p>
<ul>
<li><p>Estimated effect size <span class="math display">\[\hat\delta = \bar X_{IL} - \bar X_{c} = \hat \beta_1\]</span></p></li>
<li><p>Test statistic <span class="math display">\[
T = \frac{\bar{X}_{IL}-\bar{X}_{c}}{SD_\text{pooled} \times \sqrt{\frac{1}{n_1} + \frac{1}{n_2}}} = \frac{\hat\beta_1}{\text{SE}_{\hat\beta_1}}
\]</span></p></li>
<li><p><span class="math inline">\(\hat \beta_1\)</span> is an unbiased estimator of the real weight difference (<span class="math inline">\(\delta\)</span>) that would occur in the population of rats fed with HFD and rats fed with HFD + IL.</p></li>
</ul>
</div>
<div id="power" class="section level3">
<h3><span class="header-section-number">3.4.2</span> Power?</h3>
<p><span class="math display">\[
P[p &lt; \alpha \vert \beta_1 \neq 0]
\]</span></p>
<ul>
<li>Hence, the power will depend on the real weight difference between the group means <span class="math inline">\(\beta_1\)</span> in the population.</li>
<li>The variability of the weight measurements</li>
<li>The significance level <span class="math inline">\(\alpha\)</span></li>
<li>Sample size <span class="math inline">\(n_{IL}\)</span> and <span class="math inline">\(n_c\)</span> in both groups</li>
</ul>
<p>We can estimate the power if</p>
<ol style="list-style-type: decimal">
<li>The assumptions of the model are met: weights are normally distributed with same variance</li>
</ol>
<p>and we know</p>
<ol start="2" style="list-style-type: decimal">
<li>the standard deviation of the weight measurements around their average mean for HFD-fed mice<br />
</li>
<li>the real effect size in the population</li>
<li><span class="math inline">\(n_1\)</span> and <span class="math inline">\(n_2\)</span></li>
</ol>
</div>
<div id="use-data-from-a-previous-experiment-to-get-insight-in-mice-data" class="section level3">
<h3><span class="header-section-number">3.4.3</span> Use data from a previous experiment to get insight in mice data</h3>
<p>Suppose that we have access to the data of a preliminary experiment (e.g. provided by Karen Svenson via Gary Churchill and Dan Gatti and partially funded by P50 GM070683 on <a href="http://genomicsclass.github.io/book/pages/random_variables.html">PH525x</a>)</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>mice &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;https://raw.githubusercontent.com/genomicsclass/dagdata/master/inst/extdata/femaleMiceWeights.csv&quot;</span>)</span>
<span id="cb5-2"><a href="#cb5-2"></a></span>
<span id="cb5-3"><a href="#cb5-3"></a>mice <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>Diet,<span class="dt">y=</span>Bodyweight)) <span class="op">+</span></span>
<span id="cb5-5"><a href="#cb5-5"></a><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">outlier.shape=</span><span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="st">  </span><span class="kw">geom_jitter</span>()</span></code></pre></div>
<p><img src="experimentalDesignI_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>mice <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">sample=</span>Bodyweight)) <span class="op">+</span></span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="st">  </span><span class="kw">geom_qq</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="st">  </span><span class="kw">geom_qq_line</span>() <span class="op">+</span></span>
<span id="cb6-5"><a href="#cb6-5"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>Diet)</span></code></pre></div>
<p><img src="experimentalDesignI_files/figure-html/unnamed-chunk-2-2.png" width="672" /></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>mice &lt;-<span class="st"> </span>mice <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Diet =</span> <span class="kw">as.factor</span>(Diet))</span>
<span id="cb7-2"><a href="#cb7-2"></a>miceSum &lt;-<span class="st"> </span>mice <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="st">  </span><span class="kw">group_by</span>(Diet)  <span class="op">%&gt;%</span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="st">  </span><span class="kw">summarize_at</span>(<span class="st">&quot;Bodyweight&quot;</span>,<span class="kw">list</span>(<span class="dt">mean=</span><span class="op">~</span><span class="kw">mean</span>(.,<span class="dt">na.rm=</span><span class="ot">TRUE</span>),</span>
<span id="cb7-5"><a href="#cb7-5"></a>                    <span class="dt">sd=</span><span class="op">~</span><span class="kw">sd</span>(.,<span class="dt">na.rm=</span><span class="ot">TRUE</span>),</span>
<span id="cb7-6"><a href="#cb7-6"></a>                    <span class="dt">n=</span><span class="cf">function</span>(x) x<span class="op">%&gt;%</span>is.na<span class="op">%&gt;%</span><span class="st">`</span><span class="dt">!</span><span class="st">`</span><span class="op">%&gt;%</span>sum)) <span class="op">%&gt;%</span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">se=</span>sd<span class="op">/</span><span class="kw">sqrt</span>(n))</span>
<span id="cb7-8"><a href="#cb7-8"></a>miceSum </span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Diet"],"name":[1],"type":["fct"],"align":["left"]},{"label":["mean"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["sd"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["n"],"name":[4],"type":["int"],"align":["right"]},{"label":["se"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"chow","2":"23.81333","3":"3.022541","4":"12","5":"0.8725323"},{"1":"hf","2":"26.83417","3":"4.097606","4":"12","5":"1.1828769"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>In the experiment we have data from two diets:</p>
<ul>
<li>regular diet of cerial and grain based diet (Chow)</li>
<li>High Fat (hf)</li>
</ul>
<p>We can use the hf mice as input for our power analysis.</p>
<ul>
<li>The data of the hf mice seem to be normally distributed</li>
<li>The mean weight is 26.8g</li>
<li>The SD of the weight is 4.1g</li>
</ul>
<p>Effect size?</p>
<ul>
<li><p>The alternative hypothesis is complex.</p></li>
<li><p>It includes all possible effects!</p></li>
<li><p>In order to do the power analysis we will have to choose a minimum effect size that we would like to detect.</p></li>
<li><p>Suppose that we would like to detect if the weight of the rats difference <span class="math inline">\(\delta\)</span> of at least 10%.</p></li>
</ul>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>delta &lt;-<span class="st"> </span><span class="op">-</span><span class="st"> </span><span class="kw">round</span>(miceSum<span class="op">$</span>mean[<span class="dv">2</span>] <span class="op">*</span>.<span class="dv">1</span>,<span class="dv">1</span>)</span>
<span id="cb8-2"><a href="#cb8-2"></a>delta</span></code></pre></div>
<pre><code>## [1] -2.7</code></pre>
<p>Note, that the average weight then would get close to that of the rats in our pilot experiment that were fed on the chow diet.</p>
<p>We can set up a simulation study to assess the power of an experiment with 8 mice in each group:</p>
<div id="one-simulation" class="section level4">
<h4><span class="header-section-number">3.4.3.1</span> One simulation</h4>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="kw">set.seed</span>(<span class="dv">1423</span>)</span>
<span id="cb10-2"><a href="#cb10-2"></a>n1 &lt;-<span class="st"> </span>n2 &lt;-<span class="st"> </span><span class="dv">8</span></span>
<span id="cb10-3"><a href="#cb10-3"></a>b0 &lt;-<span class="st"> </span><span class="kw">round</span>(miceSum<span class="op">$</span>mean[<span class="dv">2</span>],<span class="dv">1</span>)</span>
<span id="cb10-4"><a href="#cb10-4"></a>b1 &lt;-<span class="st"> </span><span class="op">-</span><span class="st"> </span>delta</span>
<span id="cb10-5"><a href="#cb10-5"></a>sd &lt;-<span class="st"> </span><span class="kw">round</span>(miceSum<span class="op">$</span>sd[<span class="dv">2</span>],<span class="dv">1</span>)</span>
<span id="cb10-6"><a href="#cb10-6"></a>alpha &lt;-<span class="st"> </span><span class="fl">0.05</span></span>
<span id="cb10-7"><a href="#cb10-7"></a></span>
<span id="cb10-8"><a href="#cb10-8"></a>x &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span><span class="op">:</span><span class="dv">1</span>,<span class="kw">c</span>(n1,n2))</span>
<span id="cb10-9"><a href="#cb10-9"></a>y &lt;-<span class="st"> </span>b0 <span class="op">+</span><span class="st"> </span>b1 <span class="op">*</span><span class="st"> </span>x <span class="op">+</span><span class="st"> </span><span class="kw">rnorm</span>(n1<span class="op">+</span>n2,<span class="dv">0</span>, <span class="dt">sd =</span> sd)</span>
<span id="cb10-10"><a href="#cb10-10"></a>fit &lt;-<span class="st"> </span><span class="kw">lm</span>(y<span class="op">~</span>x)</span>
<span id="cb10-11"><a href="#cb10-11"></a>bhat &lt;-<span class="st"> </span>fit<span class="op">$</span>coef </span>
<span id="cb10-12"><a href="#cb10-12"></a>stat &lt;-<span class="st"> </span><span class="kw">summary</span>(fit)</span>
<span id="cb10-13"><a href="#cb10-13"></a><span class="kw">summary</span>(fit)<span class="op">$</span>coef[<span class="dv">2</span>,]</span></code></pre></div>
<pre><code>##   Estimate Std. Error    t value   Pr(&gt;|t|) 
##  2.5972480  1.7919658  1.4493848  0.1692595</code></pre>
<p>For this simulated experiment we would not have been able to pick up the effect of the treatment at the significance level <span class="math inline">\(\alpha\)</span>=0.05 (p = <code>r</code> round(summary(fit)$coef[2,“Pr(&gt;|t|)”],2)`)!</p>
<ul>
<li>We have to repeat the experiment many times to estimate the power.</li>
<li>We will therefore put our simulation in a general function.</li>
</ul>
</div>
<div id="simulation-of-repeated-experiments" class="section level4">
<h4><span class="header-section-number">3.4.3.2</span> Simulation of repeated experiments</h4>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="kw">library</span>(multcomp)</span></code></pre></div>
<pre><code>## Loading required package: mvtnorm</code></pre>
<pre><code>## Loading required package: survival</code></pre>
<pre><code>## Loading required package: TH.data</code></pre>
<pre><code>## Loading required package: MASS</code></pre>
<pre><code>## 
## Attaching package: &#39;MASS&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     select</code></pre>
<pre><code>## 
## Attaching package: &#39;TH.data&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:MASS&#39;:
## 
##     geyser</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a>n1 &lt;-<span class="st"> </span>n2 &lt;-<span class="st"> </span><span class="dv">8</span></span>
<span id="cb21-2"><a href="#cb21-2"></a></span>
<span id="cb21-3"><a href="#cb21-3"></a>b0 &lt;-<span class="st"> </span><span class="kw">round</span>(miceSum<span class="op">$</span>mean[<span class="dv">2</span>],<span class="dv">1</span>)</span>
<span id="cb21-4"><a href="#cb21-4"></a>b1 &lt;-<span class="st"> </span><span class="op">-</span><span class="st"> </span>delta</span>
<span id="cb21-5"><a href="#cb21-5"></a></span>
<span id="cb21-6"><a href="#cb21-6"></a>sd &lt;-<span class="st"> </span><span class="kw">round</span>(miceSum<span class="op">$</span>sd[<span class="dv">2</span>],<span class="dv">1</span>)</span>
<span id="cb21-7"><a href="#cb21-7"></a>predictorData &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Diet =</span> <span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;c&quot;</span>,<span class="st">&quot;hf&quot;</span>),<span class="kw">c</span>(n1,n2)) <span class="op">%&gt;%</span><span class="st"> </span>as.factor)</span>
<span id="cb21-8"><a href="#cb21-8"></a></span>
<span id="cb21-9"><a href="#cb21-9"></a>alpha &lt;-<span class="st"> </span><span class="fl">0.05</span></span>
<span id="cb21-10"><a href="#cb21-10"></a></span>
<span id="cb21-11"><a href="#cb21-11"></a>simLm &lt;-<span class="st"> </span><span class="cf">function</span>(form, data, betas, sd, contrasts, <span class="dt">simIndex =</span> <span class="ot">NA</span>)</span>
<span id="cb21-12"><a href="#cb21-12"></a>{</span>
<span id="cb21-13"><a href="#cb21-13"></a>  dataSim &lt;-<span class="st"> </span>data</span>
<span id="cb21-14"><a href="#cb21-14"></a>  X &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(form,dataSim)</span>
<span id="cb21-15"><a href="#cb21-15"></a>  dataSim<span class="op">$</span>ySim &lt;-<span class="st"> </span>X<span class="op">%*%</span>betas <span class="op">+</span><span class="st"> </span><span class="kw">rnorm</span>(<span class="kw">nrow</span>(dataSim),<span class="dv">0</span>,sd)</span>
<span id="cb21-16"><a href="#cb21-16"></a>  form &lt;-<span class="st"> </span><span class="kw">formula</span>(<span class="kw">paste</span>(<span class="st">&quot;ySim ~&quot;</span>,form[[<span class="dv">2</span>]]))</span>
<span id="cb21-17"><a href="#cb21-17"></a>  fitSim &lt;-<span class="st"> </span><span class="kw">lm</span>(form,dataSim)</span>
<span id="cb21-18"><a href="#cb21-18"></a>  mcp &lt;-<span class="st"> </span><span class="kw">glht</span>(fitSim,<span class="dt">linfct =</span> contrasts)</span>
<span id="cb21-19"><a href="#cb21-19"></a>  <span class="kw">return</span>(<span class="kw">summary</span>(mcp)<span class="op">$</span>test[<span class="kw">c</span>(<span class="st">&quot;coefficients&quot;</span>,<span class="st">&quot;sigma&quot;</span>,<span class="st">&quot;tstat&quot;</span>,<span class="st">&quot;pvalues&quot;</span>)]<span class="op">%&gt;%</span><span class="st"> </span>unlist)</span>
<span id="cb21-20"><a href="#cb21-20"></a>}</span>
<span id="cb21-21"><a href="#cb21-21"></a></span>
<span id="cb21-22"><a href="#cb21-22"></a><span class="kw">simLm</span>(</span>
<span id="cb21-23"><a href="#cb21-23"></a>  <span class="dt">form =</span> <span class="op">~</span>Diet,</span>
<span id="cb21-24"><a href="#cb21-24"></a>  <span class="dt">data =</span> predictorData,</span>
<span id="cb21-25"><a href="#cb21-25"></a>  <span class="dt">betas =</span> <span class="kw">c</span>(b0,b1),</span>
<span id="cb21-26"><a href="#cb21-26"></a>  <span class="dt">sd =</span> sd,</span>
<span id="cb21-27"><a href="#cb21-27"></a>  <span class="dt">contrasts =</span> <span class="st">&quot;Diethf = 0&quot;</span>)</span></code></pre></div>
<pre><code>## coefficients.Diethf        sigma.Diethf        tstat.Diethf             pvalues 
##           1.9444623           2.0931302           0.9289734           0.3686439</code></pre>
<p>We now have a generic function that can simulate data from a normal distribution for every design. The function has arguments:</p>
<ul>
<li><code>form</code>: One sided formula including the structure of the predictors in the model</li>
<li><code>data</code>: Data frame with the predictor values for the design</li>
<li><code>betas</code>: A vector with values for all mean model parameters</li>
<li><code>sd</code>: The standard deviation of the error</li>
<li><code>contrasts</code>: a scalar or a vector with the null hypotheses that we would like to assess.</li>
<li><code>simIndex</code>: an arbitrary argument that is not used by the function but that will allow it to be used in an sapply loop that runs from 1 up to the number of simulations.</li>
</ul>
<p>Simulate nSim = 1000 repeated experiments:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="kw">set.seed</span>(<span class="dv">1425</span>)</span>
<span id="cb23-2"><a href="#cb23-2"></a>nSim &lt;-<span class="st"> </span><span class="dv">1000</span></span>
<span id="cb23-3"><a href="#cb23-3"></a></span>
<span id="cb23-4"><a href="#cb23-4"></a>simResults  &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span>nSim,simLm,<span class="dt">form =</span> <span class="op">~</span>Diet,<span class="dt">data =</span> predictorData,<span class="dt">betas =</span> <span class="kw">c</span>(b0,b1),<span class="dt">sd =</span> sd,<span class="dt">contrasts =</span> <span class="st">&quot;Diethf = 0&quot;</span>))</span>
<span id="cb23-5"><a href="#cb23-5"></a>power &lt;-<span class="st"> </span><span class="kw">mean</span>(simResults[,<span class="dv">4</span>] <span class="op">&lt;</span><span class="st"> </span>alpha)</span>
<span id="cb23-6"><a href="#cb23-6"></a>power</span></code></pre></div>
<pre><code>## [1] 0.221</code></pre>
<p>We have a power of 22.1% to pick up the treatment effect when using 8 bioreps in each treatment group.</p>
</div>
<div id="power-for-multiple-sample-sizes" class="section level4">
<h4><span class="header-section-number">3.4.3.3</span> Power for multiple sample sizes</h4>
<p>We can now calculate the power for multiple sample sizes.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a>power &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">n=</span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">10</span>,<span class="dv">25</span>,<span class="dv">50</span>,<span class="dv">75</span>,<span class="dv">100</span>),<span class="dt">power=</span><span class="ot">NA</span>)</span>
<span id="cb25-2"><a href="#cb25-2"></a></span>
<span id="cb25-3"><a href="#cb25-3"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(power))</span>
<span id="cb25-4"><a href="#cb25-4"></a>{</span>
<span id="cb25-5"><a href="#cb25-5"></a>  n1 &lt;-<span class="st"> </span>n2 &lt;-<span class="st"> </span>power<span class="op">$</span>n[i]</span>
<span id="cb25-6"><a href="#cb25-6"></a>  predictorData &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Diet =</span> <span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;c&quot;</span>,<span class="st">&quot;hf&quot;</span>),<span class="kw">c</span>(n1,n2)) <span class="op">%&gt;%</span><span class="st"> </span>as.factor)</span>
<span id="cb25-7"><a href="#cb25-7"></a>  simResults  &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span>nSim,simLm,<span class="dt">form =</span> <span class="op">~</span>Diet,<span class="dt">data =</span> predictorData,<span class="dt">betas =</span> <span class="kw">c</span>(b0,b1),<span class="dt">sd =</span> sd,<span class="dt">contrasts =</span> <span class="st">&quot;Diethf = 0&quot;</span>))</span>
<span id="cb25-8"><a href="#cb25-8"></a>  power<span class="op">$</span>power[i] &lt;-<span class="st"> </span><span class="kw">mean</span>(simResults[,<span class="st">&quot;pvalues&quot;</span>] <span class="op">&lt;</span><span class="st"> </span>alpha)</span>
<span id="cb25-9"><a href="#cb25-9"></a>}</span>
<span id="cb25-10"><a href="#cb25-10"></a>power</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["n"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["power"],"name":[2],"type":["dbl"],"align":["right"]}],"data":[{"1":"3","2":"0.114"},{"1":"5","2":"0.158"},{"1":"10","2":"0.304"},{"1":"25","2":"0.648"},{"1":"50","2":"0.890"},{"1":"75","2":"0.982"},{"1":"100","2":"0.998"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a>power <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb26-2"><a href="#cb26-2"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>n,<span class="dt">y=</span>power)) <span class="op">+</span></span>
<span id="cb26-3"><a href="#cb26-3"></a><span class="st">  </span><span class="kw">geom_line</span>()</span></code></pre></div>
<p><img src="experimentalDesignI_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
</div>
<div id="assess-power-in-function-of-sample-size-for-different-effect-sizes" class="section level4">
<h4><span class="header-section-number">3.4.3.4</span> Assess power in function of sample size for different effect sizes</h4>
<p>Note, that - the sign of the delta is arbitrary because we test two-sided. - the intercept is arbitrary because we only asses <span class="math inline">\(\beta_0\)</span> - We therefore typically set <span class="math inline">\(\beta_0 = 0\)</span></p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a>nSim &lt;-<span class="st"> </span><span class="dv">1000</span></span>
<span id="cb27-2"><a href="#cb27-2"></a>b0 &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb27-3"><a href="#cb27-3"></a>deltas &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">10</span>)</span>
<span id="cb27-4"><a href="#cb27-4"></a>sd &lt;-<span class="st"> </span><span class="kw">round</span>(miceSum<span class="op">$</span>sd[<span class="dv">2</span>],<span class="dv">1</span>)</span>
<span id="cb27-5"><a href="#cb27-5"></a>ns &lt;-<span class="st">  </span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">10</span>,<span class="dv">20</span>,<span class="dv">25</span>,<span class="dv">50</span>,<span class="dv">75</span>,<span class="dv">100</span>)  </span>
<span id="cb27-6"><a href="#cb27-6"></a></span>
<span id="cb27-7"><a href="#cb27-7"></a>power &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">b1=</span><span class="kw">rep</span>(deltas,<span class="dt">each=</span><span class="kw">length</span>(ns)),</span>
<span id="cb27-8"><a href="#cb27-8"></a>                    <span class="dt">n=</span><span class="kw">rep</span>(ns,<span class="kw">length</span>(deltas)),</span>
<span id="cb27-9"><a href="#cb27-9"></a>                    <span class="dt">power=</span><span class="ot">NA</span>)</span>
<span id="cb27-10"><a href="#cb27-10"></a></span>
<span id="cb27-11"><a href="#cb27-11"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(power))</span>
<span id="cb27-12"><a href="#cb27-12"></a>{</span>
<span id="cb27-13"><a href="#cb27-13"></a>  b1 &lt;-power<span class="op">$</span>b1[i]</span>
<span id="cb27-14"><a href="#cb27-14"></a>  n1 &lt;-<span class="st"> </span>n2 &lt;-<span class="st">  </span>power<span class="op">$</span>n[i]</span>
<span id="cb27-15"><a href="#cb27-15"></a>  predictorData &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Diet =</span> <span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;c&quot;</span>,<span class="st">&quot;hf&quot;</span>),<span class="kw">c</span>(n1,n2)) <span class="op">%&gt;%</span><span class="st"> </span>as.factor)</span>
<span id="cb27-16"><a href="#cb27-16"></a>  simResults  &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span>nSim,simLm,<span class="dt">form =</span> <span class="op">~</span>Diet,<span class="dt">data =</span> predictorData,<span class="dt">betas =</span> <span class="kw">c</span>(b0,b1),<span class="dt">sd =</span> sd,<span class="dt">contrasts =</span> <span class="st">&quot;Diethf = 0&quot;</span>))</span>
<span id="cb27-17"><a href="#cb27-17"></a>  power<span class="op">$</span>power[i] &lt;-<span class="st"> </span><span class="kw">mean</span>(simResults[,<span class="st">&quot;pvalues&quot;</span>] <span class="op">&lt;</span><span class="st"> </span>alpha)</span>
<span id="cb27-18"><a href="#cb27-18"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a>power <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb28-2"><a href="#cb28-2"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>n,<span class="dt">y=</span>power,<span class="dt">col=</span>b1<span class="op">%&gt;%</span>as.factor)) <span class="op">+</span></span>
<span id="cb28-3"><a href="#cb28-3"></a><span class="st">  </span><span class="kw">geom_line</span>()</span></code></pre></div>
<p><img src="experimentalDesignI_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p>Note, that the power curves are still a bit choppy because we selected a limited number of sample sizes and because <code>nSim</code> simulations is not enough to get a good power estimate when the power is low.</p>
</div>
</div>
</div>
</div>
<div id="final-remarks" class="section level1">
<h1><span class="header-section-number">4</span> Final remarks</h1>
<ul>
<li><p>The code can be easily extended towards other designs by altering</p>
<ul>
<li>predictor data</li>
<li>formula</li>
<li>contrast</li>
</ul></li>
<li><p>The simulations start to take long when you evaluate many scenario’s.</p>
<ul>
<li>More efficient code using matrices</li>
<li>For a two group comparison closed form solutions exist</li>
</ul></li>
</ul>
<div id="more-efficient-code-based-on-matrix-algebra" class="section level2">
<h2><span class="header-section-number">4.1</span> More efficient code based on matrix algebra</h2>
<p>Code runs much faster. We now simulate 20 times more experiments in a much shorter time!</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a>nSim &lt;-<span class="st"> </span><span class="dv">20000</span></span>
<span id="cb29-2"><a href="#cb29-2"></a>b0 &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb29-3"><a href="#cb29-3"></a>sd &lt;-<span class="st"> </span><span class="kw">round</span>(miceSum<span class="op">$</span>sd[<span class="dv">2</span>],<span class="dv">1</span>)</span>
<span id="cb29-4"><a href="#cb29-4"></a>ns &lt;-<span class="st">  </span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">10</span>,<span class="dv">20</span>,<span class="dv">25</span>,<span class="dv">50</span>,<span class="dv">75</span>,<span class="dv">100</span>)  </span>
<span id="cb29-5"><a href="#cb29-5"></a>deltas &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">10</span>)</span>
<span id="cb29-6"><a href="#cb29-6"></a></span>
<span id="cb29-7"><a href="#cb29-7"></a>L &lt;-<span class="st"> </span>limma<span class="op">::</span><span class="kw">makeContrasts</span>(<span class="st">&quot;Diethf&quot;</span>,<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;(Intercept)&quot;</span>,<span class="st">&quot;Diethf&quot;</span>))</span></code></pre></div>
<pre><code>## Warning in limma::makeContrasts(&quot;Diethf&quot;, levels = c(&quot;(Intercept)&quot;, &quot;Diethf&quot;)):
## Renaming (Intercept) to Intercept</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a>powerFast &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="ot">NA</span>,<span class="dt">nrow=</span><span class="kw">length</span>(ns)<span class="op">*</span><span class="kw">length</span>(deltas),<span class="dt">ncol=</span><span class="dv">3</span>) <span class="op">%&gt;%</span><span class="st"> </span>as.data.frame</span>
<span id="cb31-2"><a href="#cb31-2"></a><span class="kw">names</span>(powerFast) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;b1&quot;</span>,<span class="st">&quot;n&quot;</span>,<span class="st">&quot;power&quot;</span>)</span>
<span id="cb31-3"><a href="#cb31-3"></a></span>
<span id="cb31-4"><a href="#cb31-4"></a>i &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb31-5"><a href="#cb31-5"></a></span>
<span id="cb31-6"><a href="#cb31-6"></a><span class="cf">for</span> (n <span class="cf">in</span> ns)</span>
<span id="cb31-7"><a href="#cb31-7"></a>{</span>
<span id="cb31-8"><a href="#cb31-8"></a>  n1 &lt;-<span class="st"> </span>n2 &lt;-<span class="st">  </span>n</span>
<span id="cb31-9"><a href="#cb31-9"></a>  </span>
<span id="cb31-10"><a href="#cb31-10"></a>  <span class="co">### Simulation</span></span>
<span id="cb31-11"><a href="#cb31-11"></a>  predictorData &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Diet =</span> <span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;c&quot;</span>,<span class="st">&quot;hf&quot;</span>),<span class="kw">c</span>(n1,n2)) <span class="op">%&gt;%</span><span class="st"> </span>as.factor)</span>
<span id="cb31-12"><a href="#cb31-12"></a>  design &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(<span class="op">~</span>Diet,predictorData)</span>
<span id="cb31-13"><a href="#cb31-13"></a>  </span>
<span id="cb31-14"><a href="#cb31-14"></a>  <span class="cf">for</span> (b1 <span class="cf">in</span> deltas)</span>
<span id="cb31-15"><a href="#cb31-15"></a>  {</span>
<span id="cb31-16"><a href="#cb31-16"></a>    ySim &lt;-<span class="st"> </span><span class="kw">rnorm</span>(<span class="kw">nrow</span>(predictorData)<span class="op">*</span>nSim,<span class="dt">sd=</span>sd)</span>
<span id="cb31-17"><a href="#cb31-17"></a>    <span class="kw">dim</span>(ySim) &lt;-<span class="kw">c</span>(<span class="kw">nrow</span>(predictorData),nSim)</span>
<span id="cb31-18"><a href="#cb31-18"></a>    ySim &lt;-<span class="st"> </span>ySim <span class="op">+</span><span class="st"> </span><span class="kw">c</span>(design <span class="op">%*%</span><span class="kw">c</span>(b0,b1))</span>
<span id="cb31-19"><a href="#cb31-19"></a>    ySim &lt;-<span class="st"> </span><span class="kw">t</span>(ySim)</span>
<span id="cb31-20"><a href="#cb31-20"></a>  </span>
<span id="cb31-21"><a href="#cb31-21"></a>    <span class="co">### Fitting</span></span>
<span id="cb31-22"><a href="#cb31-22"></a>    fitAll &lt;-<span class="st"> </span>limma<span class="op">::</span><span class="kw">lmFit</span>(ySim,design)</span>
<span id="cb31-23"><a href="#cb31-23"></a>  </span>
<span id="cb31-24"><a href="#cb31-24"></a>    <span class="co">### Inference</span></span>
<span id="cb31-25"><a href="#cb31-25"></a>    varUnscaled &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">t</span>(L)<span class="op">%*%</span>fitAll<span class="op">$</span>cov.coefficients<span class="op">%*%</span>L)</span>
<span id="cb31-26"><a href="#cb31-26"></a>    contrasts &lt;-<span class="st"> </span>fitAll<span class="op">$</span>coefficients <span class="op">%*%</span>L</span>
<span id="cb31-27"><a href="#cb31-27"></a>    seContrasts &lt;-<span class="st"> </span>varUnscaled<span class="op">^</span>.<span class="dv">5</span><span class="op">*</span>fitAll<span class="op">$</span>sigma</span>
<span id="cb31-28"><a href="#cb31-28"></a>    tstats &lt;-<span class="st"> </span>contrasts<span class="op">/</span>seContrasts</span>
<span id="cb31-29"><a href="#cb31-29"></a>    pvals &lt;-<span class="st"> </span><span class="kw">pt</span>(<span class="kw">abs</span>(tstats),fitAll<span class="op">$</span>df.residual,<span class="dt">lower.tail =</span> <span class="ot">FALSE</span>)<span class="op">*</span><span class="dv">2</span></span>
<span id="cb31-30"><a href="#cb31-30"></a>    </span>
<span id="cb31-31"><a href="#cb31-31"></a>    i &lt;-<span class="st"> </span>i<span class="op">+</span><span class="dv">1</span></span>
<span id="cb31-32"><a href="#cb31-32"></a>    powerFast[i,] &lt;-<span class="st"> </span><span class="kw">c</span>(b1,n,<span class="kw">mean</span>(pvals <span class="op">&lt;</span><span class="st"> </span>alpha))</span>
<span id="cb31-33"><a href="#cb31-33"></a>  }</span>
<span id="cb31-34"><a href="#cb31-34"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a>powerFast <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb32-2"><a href="#cb32-2"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>n,<span class="dt">y=</span>power,<span class="dt">col=</span>b1<span class="op">%&gt;%</span>as.factor)) <span class="op">+</span></span>
<span id="cb32-3"><a href="#cb32-3"></a><span class="st">  </span><span class="kw">geom_line</span>()</span></code></pre></div>
<p><img src="experimentalDesignI_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
</div>
<div id="more-efficient-code-based-on-closed-form-solution-that-exists-for-two-group-comparison" class="section level2">
<h2><span class="header-section-number">4.2</span> More efficient code based on closed form solution that exists for two group comparison</h2>
<p>For the two sample t-test a closed form estimate exists for the power. In this context the Cohen’s effect size is typically used:</p>
<p><span class="math inline">\(D = \frac{\delta}{SD}\)</span></p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a>D &lt;-<span class="st"> </span>delta<span class="op">/</span>sd</span>
<span id="cb33-2"><a href="#cb33-2"></a><span class="kw">library</span>(pwr)</span>
<span id="cb33-3"><a href="#cb33-3"></a><span class="kw">pwr.t.test</span>(<span class="dt">n=</span><span class="dv">8</span>, <span class="dt">d=</span>D, <span class="dt">type=</span><span class="st">&#39;two.sample&#39;</span>)</span></code></pre></div>
<pre><code>## 
##      Two-sample t test power calculation 
## 
##               n = 8
##               d = 0.6585366
##       sig.level = 0.05
##           power = 0.2327553
##     alternative = two.sided
## 
## NOTE: n is number in *each* group</code></pre>
<p>Note, that this is very similar to the power that we calculated using the simulations!</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a>b0 &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb35-2"><a href="#cb35-2"></a>sd &lt;-<span class="st"> </span><span class="kw">round</span>(miceSum<span class="op">$</span>sd[<span class="dv">2</span>],<span class="dv">1</span>)</span>
<span id="cb35-3"><a href="#cb35-3"></a>ns &lt;-<span class="st">  </span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">10</span>,<span class="dv">20</span>,<span class="dv">25</span>,<span class="dv">50</span>,<span class="dv">75</span>,<span class="dv">100</span>)  </span>
<span id="cb35-4"><a href="#cb35-4"></a>Ds &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">10</span>)<span class="op">/</span>sd</span>
<span id="cb35-5"><a href="#cb35-5"></a></span>
<span id="cb35-6"><a href="#cb35-6"></a></span>
<span id="cb35-7"><a href="#cb35-7"></a>powerTheo &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Ds=</span><span class="kw">rep</span>(Ds,<span class="dt">each=</span><span class="kw">length</span>(ns)),</span>
<span id="cb35-8"><a href="#cb35-8"></a>                    <span class="dt">n=</span><span class="kw">rep</span>(ns,<span class="kw">length</span>(Ds)),</span>
<span id="cb35-9"><a href="#cb35-9"></a>                    <span class="dt">power=</span><span class="ot">NA</span>)</span>
<span id="cb35-10"><a href="#cb35-10"></a></span>
<span id="cb35-11"><a href="#cb35-11"></a>powerTheo<span class="op">$</span>power &lt;-<span class="st"> </span><span class="kw">apply</span>(powerTheo[,<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>],<span class="dv">1</span>,<span class="cf">function</span>(x) <span class="kw">pwr.t.test</span>(<span class="dt">d=</span>x[<span class="dv">1</span>],<span class="dt">n=</span>x[<span class="dv">2</span>],<span class="dt">type=</span><span class="st">&quot;two.sample&quot;</span>)<span class="op">$</span>power)</span></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a>powerTheo <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb36-2"><a href="#cb36-2"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>n,<span class="dt">y=</span>power,<span class="dt">col=</span>Ds<span class="op">%&gt;%</span>as.factor)) <span class="op">+</span></span>
<span id="cb36-3"><a href="#cb36-3"></a><span class="st">  </span><span class="kw">geom_line</span>()</span></code></pre></div>
<p><img src="experimentalDesignI_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
</div>
</div>

<div id="rmd-source-code">---
title: "Experimental Design I: replicatie en power"
author: "Lieven Clement"
date: "statOmics, Ghent University (https://statomics.github.io)"
output:
  html_document:
    code_download: yes
    theme: cosmo
    toc: yes
    toc_float: yes
    highlight: tango
    number_sections: yes
---


<a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a>

```{r}
library(tidyverse)
```

# Concepts

- Experimental units representative for population: Randomisation 
- Replication: technical vs biological, sample size - power
- Sources of variation: technical, biological, within subject, between subject

# Replicatie 

Two pager on Replication in nature methods [[PDF](https://www.nature.com/articles/nmeth.3091.pdf)]
![](https://www.nature.com/articles/nmeth.3091.pdf){ width=100% }

## Example

Consider a single cell RNA-seq experiment where researchers want to assess the effect of drug treatment on gene expression

Potential research questions

- Is there a difference in average gene expression between treated and non treated samples
- Is there a difference in variability of gene expression between treated and non treated samples 

## Sources of variation

TABLE 1 NATURE METHODS | VOL.11 NO.9 | SEPTEMBER 2014 | 879 - 880

|   | Replicate type | Replicate category$^\text{a}$ |
|:---:|:---|:---:|
| | Colonies |	B |
| | Strains |	B |
| Animal study subjects | Cohoused groups |	B |
| | Gender | B |
| | Individuals	| B | 
| | | |
| | Organs from sacrificed animals | B |
| | Methods for dissociating cells from tissue | T |
| Sample preparation | Dissociation runs from given tissue sample | T |
| | Individual cells | B |
| | | |
| | RNA-seq library construction | T |
| | Runs from the library of a given cell |	T |
| Sequencing | Reads from different transcript molecules | V$^\text{b}$ |
| | Reads with unique molecular identifier (UMI) from a given transcript molecule |	T |

(a) Replicates are categorized as biological (B), technical (T) or of variable type (V). 
(b) Sequence reads serve diverse purposes depending on the application and how reads are used in analysis.

---

## At which level should we replicate?

- $\text{var}(X)=\sigma^2_A+\sigma^2_C+\sigma^2_M=\sigma^2_{TOT}$
- $\text{var}(\bar{X})=\frac{\sigma^2_A}{n_A}+\frac{\sigma^2_C}{n_A n_C} + \frac{\sigma^2_M}{n_A n_C n_M}$

![](https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fnmeth.3091/MediaObjects/41592_2014_Article_BFnmeth3091_Fig1_HTML.jpg)
Figure 1 NATURE METHODS | VOL.11 NO.9 | SEPTEMBER 2014 | 879 - 880

(a) Three levels of replication (two biological, one technical) with animal, cell and measurement replicates normally distributed with a mean across animals of 10 and ratio of variances 1:2:0.5. Solid green (biological) and blue (technical) dots show how a measurement of the expression (X = 12) samples from all three sources of variation. Distribution s.d. is shown as horizontal lines. 
(b) Expression variance, Var(X), and variance of expression mean, Var($\bar X$), computed across 10,000 simulations of nAnCnM = 48 measurements for unique combinations of the number of animals (nA = 1 to 48), cells per animal (nC = 1 to 48) and technical replicate measurements per cell (nM = 1 and 3). The ratio of Var(X) and Var($\bar X$) is the effective sample size, n, which corresponds to the equivalent number of statistically independent measurements. Horizontal dashed lines correspond to biological and total variation. Error bars on Var(X) show s.d. from the 10,000 simulated samples (nM = 1).

---

- Cost benefit trade-off: cost difference between biological and technical replicates 
- Typically, biological variability is substantially larger than technical variability

$\rightarrow$ commit resources to sampling biologically relevant variables 

$\rightarrow$ unless measures of technical variability are of interest then increasing number of technical measurements is valuable.

- Good experimental design practice includes planning for replication. 

  1. Identify the research questions to be answered with experiments. 
  2. Determine proportion of variability induced by each step.
  3. Be aware for pseudoreplication and aim for statistically independent replicates.

--- 

# Power, sample size and other design aspects

Reading materials: 
[Nature Methods (2013), 10(12), 1139–1140](https://www.nature.com/articles/nmeth.2738.pdf)

## Intermezzo linear regression in matrix form 

- Linear regression is a very important statistical tool to study the association between variables and to build prediction models.

### Scalar form

- Consider a vector of predictors $\mathbf{x}=(x_1,\ldots,x_p)$ and
- a real-valued response $Y$
- then the linear regression model can be written as
\[
Y=f(\mathbf{x}) +\epsilon=\beta_0+\sum\limits_{j=1}^p x_j\beta_j + \epsilon
\]
with i.i.d. $\epsilon\sim N(0,\sigma^2)$



### Vector/Matrix form

- $n$ observations $(\mathbf{x}_1,y_1) \ldots (\mathbf{x}_n,y_n)$ with $\mathbf{x}_1^T=[1 x_1 \ldots x_p]$
- Regression in matrix notation
\[\mathbf{Y}=\mathbf{X\beta} + \mathbf{\epsilon}\]
with $\mathbf{Y}=\left[\begin{array}{c}y_1\\ \vdots\\y_n\end{array}\right]$,
$\mathbf{X}=\left[\begin{array}{cccc} 1&x_{11}&\ldots&x_{1p}\\
\vdots&\vdots&&\vdots\\
1&x_{n1}&\ldots&x_{np}
\end{array}\right]$ or $\mathbf{X}=\left[\begin{array}{c} \mathbf{x}_1^T\\\vdots\\\mathbf{x}_n^T\end{array}\right]$,
$\boldsymbol{\beta}=\left[\begin{array}{c}\beta_0\\ \vdots\\ \beta_p\end{array}\right]$ and
$\mathbf{\epsilon}=\left[\begin{array}{c} \epsilon_1 \\ \vdots \\ \epsilon_n\end{array}\right]$

---

## Least Squares (LS)

- Minimize the residual sum of squares
\begin{eqnarray*}
RSS(\boldsymbol{\beta})&=&\sum\limits_{i=1}^n e^2_i\\
&=&\sum\limits_{i=1}^n \left(y_i-\beta_0-\sum\limits_{j=1}^p x_{ij}\beta_j\right)^2
\end{eqnarray*}

- or in matrix notation

$$
\text{RSS}(\boldsymbol{\beta})=(\mathbf{Y}-\mathbf{X\beta})^T(\mathbf{Y}-\mathbf{X\beta})
$$

$$\rightarrow \hat{\boldsymbol{\beta}}=\text{argmin}_\beta \text{ RSS}(\boldsymbol{\beta})$$

---

### Minimize RSS
\[
\begin{array}{ccc}
\frac{\partial RSS}{\partial \boldsymbol{\beta}}&=&\mathbf{0}\\\\
\frac{(\mathbf{Y}-\mathbf{X\beta})^T(\mathbf{Y}-\mathbf{X}\boldsymbol{\beta})}{\partial \boldsymbol{\beta}}&=&\mathbf{0}\\\\
-2\mathbf{X}^T(\mathbf{Y}-\mathbf{X}\boldsymbol{\beta})&=&\mathbf{0}\\\\
\mathbf{X}^T\mathbf{X\beta}&=&\mathbf{X}^T\mathbf{Y}\\\\
\hat{\boldsymbol{\beta}}&=&(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{Y}
\end{array}
\]

---

## Variance Estimator?
\[
\begin{array}{ccl}
\hat{\boldsymbol{\Sigma}}_{\hat{\boldsymbol{\beta}}}
&=&\text{var}\left[(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{Y}\right]\\\\
&=&(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\text{var}\left[\mathbf{Y}\right]\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}\\\\
&=&(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T(\mathbf{I}\sigma^2)\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}
\\\\
&=&(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{I}\quad\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}\sigma^2\\\\
%\hat{\boldmath{\Sigma}}_{\hat{\boldsymbol{\beta}}}&=&(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T \text{var}\left[\mathbf{Y}\right](\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}\\
&=&(\mathbf{X}^T\mathbf{X})^{-1}\mathbf{X}^T\mathbf{X}(\mathbf{X}^T\mathbf{X})^{-1}\sigma^2\\\\
&=&(\mathbf{X}^T\mathbf{X})^{-1}\sigma^2
\end{array}
\]

The uncertainty on the model parameters thus depends on the residual variability and the design! 

- The larger $\mathbf{X}^T\mathbf{X}$ the more information the experiment contains on the model parameters and the smaller their variances and standard errors will be! 
- Factorial designs? 
- Designs with continuous predictors? 

---

The effect size of interest is typically a linear combinations of the model parameters, i.e. 
$$
l_0 \times \beta_0 + l_1 \times \beta_1 + ... + l_{p-1} \times \beta_{p-1} = \mathbf{L}^T\boldsymbol{\beta}
$$ 

The null hypothesis of our test statistics is than formulated as 


$$
H_0: \mathbf{L}^T\boldsymbol{\beta} = 0 
$$ 

vs the alternative hypothesis 

$$
H_0: \mathbf{L}^T\boldsymbol{\beta} \neq 0 
$$
which can be assessed using a t-test statistic: 

$$
t=\frac{\mathbf{L}^T\hat{\boldsymbol{\beta}} - 0}{\text{se}_{\mathbf{L}^T\hat{\boldsymbol{\beta}}}}
$$
which follows a t-distribution with n-p degrees of freedom under the null hypothesis when all assumptions for the linear model are valid. 

So the power 

$$P(p < 0.05 | H_1)$$

will typically depends on 

- the real effect size in the population  $\mathbf{L}^T\boldsymbol{\beta}$.
- Number of observations: SE and df of t-test.
- Choice of the design points.
- Choice of significance level. 

Similar to the introductory example, we can use simulations to assess the power.

## Mouse example 

- In 2021 Choa et al. published that the cytokine Thymic stromal lymphopoietin
(TSLP) induced fat loss through sebum secretion (talg). [[html](https://www.science.org/doi/full/10.1126/science.abd2893)] [[PDF](https://www.science.org/doi/pdf/10.1126/science.abd2893)]

- Suppose that you would like to set up a similar study to test if cytokine interleukin 25 (IL) also has beneficial effect. 

- You plan to setup a study with a control group of high fat diet (HFD) fed mice and a treatment group that recieves the HFD and IL. 

- What sample size do you need to pick up the effect of the treatment. 


### How will we analyse the data of this experiment? 


---



- $H_0$: The average weight difference is equal to zero
- $H-0$: The average weight difference is different from zero

- Two sample t-test or a t-test on the slope of a linear model with one dummy variable. 

$$ Y_i = \beta_0 + \beta_1 X_\text{iIL} + \epsilon_i$$ 

$$
\text{with }
X_\text{iIL}=\begin{cases}X_{iIL}=0 & \text{HFD}\\X_{iIL}=1 & \text{HFD + IL}\end{cases}.
$$

- Estimated effect size 
$$\hat\delta = \bar X_{IL} - \bar X_{c} = \hat \beta_1$$

- Test statistic
$$
T = \frac{\bar{X}_{IL}-\bar{X}_{c}}{SD_\text{pooled} \times \sqrt{\frac{1}{n_1} + \frac{1}{n_2}}} = \frac{\hat\beta_1}{\text{SE}_{\hat\beta_1}}
$$
- $\hat \beta_1$ is an unbiased estimator of the real weight difference ($\delta$) that would occur in the population of rats fed with HFD and rats fed with HFD + IL.  

### Power? 

$$
P[p < \alpha \vert \beta_1 \neq 0]
$$


- Hence, the power will depend on the real weight difference between the group means $\beta_1$ in the population.
- The variability of the weight measurements
- The significance level $\alpha$
- Sample size $n_{IL}$ and $n_c$ in both groups 


We can estimate the power if 

1. The assumptions of the model are met: weights are normally distributed with same variance

and we know

2. the standard deviation of the weight measurements around their average mean for HFD-fed mice  
3. the real effect size in the population
4. $n_1$ and $n_2$ 

### Use data from a previous experiment to get insight in mice data

Suppose that we have access to the data of a preliminary experiment (e.g. provided by Karen Svenson via Gary Churchill and Dan Gatti and partially funded by P50 GM070683 on [PH525x](http://genomicsclass.github.io/book/pages/random_variables.html))

```{r}
mice <- read.csv("https://raw.githubusercontent.com/genomicsclass/dagdata/master/inst/extdata/femaleMiceWeights.csv")

mice %>% 
  ggplot(aes(x=Diet,y=Bodyweight)) +
  geom_boxplot(outlier.shape=FALSE) +
  geom_jitter()

mice %>% 
  ggplot(aes(sample=Bodyweight)) +
  geom_qq() + 
  geom_qq_line() +
  facet_wrap(~Diet)
```

```{r}
mice <- mice %>% mutate(Diet = as.factor(Diet))
miceSum <- mice %>% 
  group_by(Diet)  %>%
  summarize_at("Bodyweight",list(mean=~mean(.,na.rm=TRUE),
                    sd=~sd(.,na.rm=TRUE),
                    n=function(x) x%>%is.na%>%`!`%>%sum)) %>%
  mutate(se=sd/sqrt(n))
miceSum 
```
In the experiment we have data from two diets: 

- regular diet of cerial and grain based diet (Chow) 
- High Fat (hf)

We can use the hf mice as input for our power analysis. 

- The data of the hf mice seem to be normally distributed 
- The mean weight is `r miceSum %>% filter(Diet == "hf") %>% pull("mean") %>% round(.,1)`g 
- The SD of the weight is `r miceSum %>% filter(Diet == "hf") %>% pull("sd") %>% round(.,1)`g 

Effect size? 

- The alternative hypothesis is complex. 
- It includes all possible effects! 
- In order to do the power analysis we will have to choose a minimum effect size that we would like to detect. 

- Suppose that we would like to detect if the weight of the rats difference $\delta$ of at least 10%. 

```{r}
delta <- - round(miceSum$mean[2] *.1,1)
delta
```

Note, that the average weight then would get close to that of the rats in our pilot experiment that were fed on the chow diet.

We can set up a simulation study to assess the power of an experiment with 8 mice in each group:

#### One simulation

```{r}
set.seed(1423)
n1 <- n2 <- 8
b0 <- round(miceSum$mean[2],1)
b1 <- - delta
sd <- round(miceSum$sd[2],1)
alpha <- 0.05

x <- rep(0:1,c(n1,n2))
y <- b0 + b1 * x + rnorm(n1+n2,0, sd = sd)
fit <- lm(y~x)
bhat <- fit$coef 
stat <- summary(fit)
summary(fit)$coef[2,]
```

For this simulated experiment we would `r ifelse(summary(fit)$coef[2,"Pr(>|t|)"] < alpha,"","not")` have been able to pick up the effect of the treatment at the significance level $\alpha$=`r alpha` (p = `r` round(summary(fit)$coef[2,"Pr(>|t|)"],2)`)! 

- We have to repeat the experiment many times to estimate the power. 
- We will therefore put our simulation in a general function. 

#### Simulation of repeated experiments

```{r}
library(multcomp)
n1 <- n2 <- 8

b0 <- round(miceSum$mean[2],1)
b1 <- - delta

sd <- round(miceSum$sd[2],1)
predictorData <- data.frame(Diet = rep(c("c","hf"),c(n1,n2)) %>% as.factor)

alpha <- 0.05

simLm <- function(form, data, betas, sd, contrasts, simIndex = NA)
{
  dataSim <- data
  X <- model.matrix(form,dataSim)
  dataSim$ySim <- X%*%betas + rnorm(nrow(dataSim),0,sd)
  form <- formula(paste("ySim ~",form[[2]]))
  fitSim <- lm(form,dataSim)
  mcp <- glht(fitSim,linfct = contrasts)
  return(summary(mcp)$test[c("coefficients","sigma","tstat","pvalues")]%>% unlist)
}

simLm(
  form = ~Diet,
  data = predictorData,
  betas = c(b0,b1),
  sd = sd,
  contrasts = "Diethf = 0")
```

We now have a generic function that can simulate data from a normal distribution for every design. 
The function has arguments: 

- `form`: One sided formula including the structure of the predictors in the model 
- `data`: Data frame with the predictor values for the design
- `betas`: A vector with values for all mean model parameters
- `sd`: The standard deviation of the error 
- `contrasts`: a scalar or a vector with the null hypotheses that we would like to assess. 
- `simIndex`: an arbitrary argument that is not used by the  function but that will allow it to be used in an sapply loop that runs from 1 up to the number of simulations. 

Simulate nSim = 1000 repeated experiments: 

```{r}
set.seed(1425)
nSim <- 1000

simResults  <- t(sapply(1:nSim,simLm,form = ~Diet,data = predictorData,betas = c(b0,b1),sd = sd,contrasts = "Diethf = 0"))
power <- mean(simResults[,4] < alpha)
power
```

We have a power of `r power*100`% to pick up the treatment effect when using 8 bioreps in each treatment group. 

#### Power for multiple sample sizes 

We can now calculate the power for multiple sample sizes. 

```{r}
power <- data.frame(n=c(3,5,10,25,50,75,100),power=NA)

for (i in 1:nrow(power))
{
  n1 <- n2 <- power$n[i]
  predictorData <- data.frame(Diet = rep(c("c","hf"),c(n1,n2)) %>% as.factor)
  simResults  <- t(sapply(1:nSim,simLm,form = ~Diet,data = predictorData,betas = c(b0,b1),sd = sd,contrasts = "Diethf = 0"))
  power$power[i] <- mean(simResults[,"pvalues"] < alpha)
}
power
```

```{r}
power %>% 
  ggplot(aes(x=n,y=power)) +
  geom_line()
```

#### Assess power in function of sample size for different effect sizes 

Note, that 
- the sign of the delta is arbitrary because we test two-sided. 
- the intercept is arbitrary because we only asses $\beta_0$
- We therefore typically set $\beta_0 = 0$

```{r}
nSim <- 1000
b0 <- 0
deltas <- c(1,2,3,5,10)
sd <- round(miceSum$sd[2],1)
ns <-  c(3,5,10,20,25,50,75,100)  

power <- data.frame(b1=rep(deltas,each=length(ns)),
                    n=rep(ns,length(deltas)),
                    power=NA)

for (i in 1:nrow(power))
{
  b1 <-power$b1[i]
  n1 <- n2 <-  power$n[i]
  predictorData <- data.frame(Diet = rep(c("c","hf"),c(n1,n2)) %>% as.factor)
  simResults  <- t(sapply(1:nSim,simLm,form = ~Diet,data = predictorData,betas = c(b0,b1),sd = sd,contrasts = "Diethf = 0"))
  power$power[i] <- mean(simResults[,"pvalues"] < alpha)
}
```

```{r}
power %>% 
  ggplot(aes(x=n,y=power,col=b1%>%as.factor)) +
  geom_line()
```

Note, that the power curves are still a bit choppy because we selected a limited number of sample sizes and because `nSim` simulations is not enough to get a good power estimate when the power is low. 

# Final remarks

- The code can be easily extended towards other designs by altering
    
    - predictor data 
    - formula
    - contrast

- The simulations start to take long when you evaluate many scenario's. 
    
    - More efficient code using matrices
    - For a two group comparison closed form solutions exist
    



## More efficient code based on matrix algebra

Code runs much faster. We now simulate 20 times more experiments in a much shorter time! 

```{r}
nSim <- 20000
b0 <- 0
sd <- round(miceSum$sd[2],1)
ns <-  c(3,5,10,20,25,50,75,100)  
deltas <- c(1,2,3,5,10)

L <- limma::makeContrasts("Diethf",levels=c("(Intercept)","Diethf"))

powerFast <- matrix(NA,nrow=length(ns)*length(deltas),ncol=3) %>% as.data.frame
names(powerFast) <- c("b1","n","power")

i <- 0

for (n in ns)
{
  n1 <- n2 <-  n
  
  ### Simulation
  predictorData <- data.frame(Diet = rep(c("c","hf"),c(n1,n2)) %>% as.factor)
  design <- model.matrix(~Diet,predictorData)
  
  for (b1 in deltas)
  {
    ySim <- rnorm(nrow(predictorData)*nSim,sd=sd)
    dim(ySim) <-c(nrow(predictorData),nSim)
    ySim <- ySim + c(design %*%c(b0,b1))
    ySim <- t(ySim)
  
    ### Fitting
    fitAll <- limma::lmFit(ySim,design)
  
    ### Inference
    varUnscaled <- c(t(L)%*%fitAll$cov.coefficients%*%L)
    contrasts <- fitAll$coefficients %*%L
    seContrasts <- varUnscaled^.5*fitAll$sigma
    tstats <- contrasts/seContrasts
    pvals <- pt(abs(tstats),fitAll$df.residual,lower.tail = FALSE)*2
    
    i <- i+1
    powerFast[i,] <- c(b1,n,mean(pvals < alpha))
  }
}
```

```{r}
powerFast %>% 
  ggplot(aes(x=n,y=power,col=b1%>%as.factor)) +
  geom_line()
```

## More efficient code based on closed form solution that exists for two group comparison

For the two sample t-test a closed form estimate exists for the power. 
In this context the Cohen's effect size is typically used: 

$D = \frac{\delta}{SD}$


```{r}
D <- delta/sd
library(pwr)
pwr.t.test(n=8, d=D, type='two.sample')
```
Note, that this is very similar to the power that we calculated using the simulations! 

```{r}
b0 <- 0
sd <- round(miceSum$sd[2],1)
ns <-  c(3,5,10,20,25,50,75,100)  
Ds <- c(1,2,3,5,10)/sd


powerTheo <- data.frame(Ds=rep(Ds,each=length(ns)),
                    n=rep(ns,length(Ds)),
                    power=NA)

powerTheo$power <- apply(powerTheo[,1:2],1,function(x) pwr.t.test(d=x[1],n=x[2],type="two.sample")$power)
```


```{r}
powerTheo %>% 
  ggplot(aes(x=n,y=power,col=Ds%>%as.factor)) +
  geom_line()
```

</div>
<div class="footer">
    <hr>
    This work is licensed under the <a href= "https://creativecommons.org/licenses/by-nc-sa/4.0">
    CC BY-NC-SA 4.0</a> licence.
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("experimentalDesignI.Rmd");
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
